src/test/sighash_tests.cpp

```diff
@@ -244,6 +244,256 @@ BOOST_AUTO_TEST_CASE(sighash_from_data)
 
         sh = SignatureHash(scriptCode, tx, nIn, nHashType, 0, tx.GetRequiredSigVersion());
         BOOST_CHECK_MESSAGE(sh.GetHex() == sigHashHex, strTest);
+
+    }
+}
+
+BOOST_AUTO_TEST_CASE(malleated_tx)
+{
+    int nRandomTests = 5000;
+    std::vector<uint256> vsh;
+    std::vector<CScript> vScriptCode;
+    for (int t = 0; t < nRandomTests; t++) {
+        vsh.clear();
+        vScriptCode.clear();
+        // create a random tx and get the signature hashes
+        CMutableTransaction _tx;
+        RandomTransaction(_tx, false);
+        const CTransaction tx(_tx);
+        for (int nIn = 0; nIn < (int) tx.vin.size(); nIn++) {
+            vScriptCode.emplace_back();
+            RandomScript(vScriptCode.back());
+            vsh.emplace_back(SignatureHash(vScriptCode.back(), tx, nIn, SIGHASH_ALL, 0, tx.GetRequiredSigVersion()));
+        }
+
+        // -- create malleated txes
+        CMutableTransaction mtx = _tx;
+        for (int nIn = 0; nIn < (int) tx.vin.size(); nIn++) {
+            BOOST_CHECK(vsh[nIn] == SignatureHash(vScriptCode[nIn], CTransaction(mtx), nIn, SIGHASH_ALL, 0, tx.GetRequiredSigVersion()));
+        }
+
+        // remove one transparent input
+        if (mtx.vin.size() > 0) {
+            mtx.vin.pop_back();
+            CTransaction tx2(mtx);
+            for (int nIn = 0; nIn < (int) tx.vin.size() - 1; nIn++) {
+                BOOST_CHECK(vsh[nIn] != SignatureHash(vScriptCode[nIn], tx2, nIn, SIGHASH_ALL, 0, tx2.GetRequiredSigVersion()));
```

Comment:
would be good to use `BOOST_CHECK_MESSAGE` to be more precise on the error description (same for the rest).

Reply:
I think it's fine. They are all checking the same thing.